### ISSUE TITLE
As non-DataCurator user, method changes not noticed

### ISSUE THREAD

**Issue by LisaAlmarode** [dalehenrich/Rowan#227]
I created a new user with full permissions, and loaded RowanSample4.

Changes to existing methods, adding new methods do not show up in changes and do not get written to disk.  An added class does show up in changes and gets a new file created on write (but no methods are there).

Reported in HR 9824

**Comment by dalehenrich** [comment]
The method is being changed in the class, but the loaded method is not updated...

**Comment by dalehenrich** [comment]
Interestingly enough the following executed in a workspace, does work:
```smalltalk
RowanSample1Test
	rwCompileMethod: 
'testFailure

	self assert: RowanSample1 new foo = ''car''
'
	category: 'tests'
```
... and this is pretty much the code that *should* be execute by Jadeite???

**Comment by dalehenrich** [comment]
... okay this is a *known bug*: issue #177, #178, and #205. The following expression needs to be evaluated in the context of a user using Jadeite:
```smalltalk
GsPackagePolicy current enable.
UserGlobals at: #rowanCompile put: true.
System commit
```
We automatically do this for `DataCurator` and `SystemUser` when installing Rowan ... I suppose we could do this for **AllUsers** when we install --- what do you think @LisaAlmarode ...

**Comment by LisaAlmarode** [comment]
Has to be done for all users... otherwise other users can't do stuff to methods, which appears to be a requirement.

**Comment by dalehenrich** [comment]
Yeah, but it does not fix the problem if users are added after Rowan has been installed ... and the release  version of Jadeite (pyrite) will not have this requirement ...

**Comment by dalehenrich** [comment]
I'm inclined to have the user chime in on this one --- they are the only ones who will see it ...

**Comment by LisaAlmarode** [comment]
Maybe a login hook or something.  This is something that no one is going to remember since users are not often created; and it causes a hard-to-see problem... one does not want to take chances with code getting lost.

**Comment by dalehenrich** [comment]
Yes but we have a captive audience ... this is a short term problem that might be solved by a one-time doit on the customer end ... which is why I want their opinion ... this in no way is a long term workaround (unless you call 4 weeks long term:)

**Comment by LisaAlmarode** [comment]
Why didn't you think you should do it for all users on rowan load, then?  

Anyway... you asked and you have my opinion...  I like safe code.

**Comment by dalehenrich** [comment]
A login hook would require a new release of Jadite and that is not something that will happen in the next week ... Eric should be spending his time working on the release version ... I think it is likely to be a onetime patch for the customer --- if we ask/remind them ... they can add the three lines to their installation script and make sure that the development suers are covered ...

**Comment by dalehenrich** [comment]
give me a call and discuss this ... we can dispose of this in two minutes ...

**Comment by LisaAlmarode** [comment]
Here's what I think: let's not get too cute about making sure all users get this code change.  

The real goal is to make sure that if you do something like compile a method and write the project, the change definitely positively is in fact written to disk and won't get mysteriously lost if you reload.

So... make it an error to compile, if say, rowanCompile isn't true, or whatever the correct tests is.  That way you'd know something was wrong and have a chance to find the magic incantation to satisfy Rowan.

**Comment by dalehenrich** [comment]
... nice try, but the reason that the global is to explicitly **allow** someone to compile a non-Rowan method ... if it was simple, it would be done by now ... let's talk it over on phone or when we're both in office

**Comment by dalehenrich** [comment]
With the completion of #364, we'll be ready to remove #rowanCompile from Jadeite, the Rowan install scripts and docs
